Initial routes adjustment

mentalcaries opened the pull request:
## Description

For this PR I made adjustments to the main routes based on @5hraddha's suggestions and practices I've noted on other sites.

- The main landing page is now corporate user facing (the content that was previously at `/admin` is now on the main page.
- The route to login is now `/admin/login` and the header buttons have been updated to suit
- The dashboard remains at `/admin/dashboard`
- `/admin` remains a protected route, so users are redirected based on the presence of a valid session
- A dynamic route now exists for practices at `/[practiceCode]` and this can be set up for future use for practice profiles/information as @5hraddha had proposed
- The appointment pages are now at `/[practiceCode]/book`. I thought using `book` was shorter and easier than `appointment`


## Related Issue

<!-- If you write "closes" followed by the Github issue number, it will automatically close the issue for you when the PR merges -->

## Acceptance Criteria

<!-- Include AC from the Github issue -->

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

<!-- If UI feature, take provide screenshots -->

### After

<!-- If UI feature, take provide screenshots -->

## Testing Steps / QA Criteria

5hraddha reviewed (commented):
Hi @mentalcaries,

Good job in refactoring the routes. 😎 It looks great! I like the `/book` url too. It's short and clean 🚀

There are a few things to check:
1. Once logged in as an admin if I navigate to the home page `http://localhost:3000/`, the header still shows **Login**. It should be toggled to **Logout** instead when the admin is already logged in.

2. You still have to implement a dynamic route for practices at `/[practiceCode]` right? Like - http://localhost:3000/pearly-whites. Please, correct me if I'm missing something. I couldn't find that in code either. But, since you were referring in the PR, I'm confused.

**A good reference that I found**
I also saw this website today - [https://gun.io/](https://gun.io/). They have two groups of customers too. The hiring guys and the ones looking for jobs. I like how they have their menu in the header: 
<img width="1669" alt="image" src="https://github.com/Full-Stack-Collective/connectient/assets/27571141/8c9d0b97-98c0-4eba-ba94-8ce005ec92af">

The main home page is more admin centered. They have a link in the header - `Find Work` that takes to a landing page for job seekers.

They also have `Pricing` and `About Us` links. We would need that sometime soon. 

The main landing page - [https://gun.io/](https://gun.io/) that few interesting things:
- `What we do` section
- `See what our clients have to say` section
- `How does it work?` (in clear steps) section - I like this one as it clearly states the process
- `See what Gun.io can do for you` section for contact
- And finally their footer with the parent company's logo, social media details, a way for folks to reach out, and separate links like - [Find Talent](https://gun.io/find-talent/), [Find Work](https://gun.io/find-work/), [FAQ](https://gun.io/faq/), etc.

The similar stuffs have beem clearly portrayed in [https://gun.io/find-work/](https://gun.io/find-work/) from the developers perspective. 

We need to come up with similar stuffs for our website. I'll catch up with you offline on this and see what do you think!

mentalcaries commented:
@5hraddha I still haven't done much for the dynamic route for the practices till we decide what we're doing with it. Maybe create a profile page. I'm not focused on it because it's not relevant to the appointment form.

I made some changes to the`Header` component. I updated the `getAuthenticationMenuLink `. I extracted it to the constants.ts file and renamed it to getNavigationLinks. That function now renders the navigation items based on whether there's a valid session or not and is used in both the landing page and dashboard views.
Not sure if constants.ts is the best place for it, but it works for now. I also cleaned up some of the unused constants because they're just adding clutter

Lastly, I created a separate `AppointmentHeader` component for the appointment form. It utilizes the name logo of the relevant Practice instead of Connectient. Our Connectient branding remains at the footer though for the "Powered by..."

5hraddha reviewed (approved):
@mentalcaries Looks good! 🚀
